### PR TITLE
[CORL-149] Persisted Queries

### PR DESCRIPTION
**Pull request by alinhle**
## What does this PR do?
- Enabled persisted queries for the Client
- `npm run build` will build using persisted queries
- `npm run generate-persist` and `npm run generate:relay-persist` will also generate artefacts with persisted queries

## TODO
- [x] Server implementation

### CODE REVIEW

File: src/core/client/framework/lib/network/createManagedSubscriptionClient.ts [hunk 1]
```diff
@@ -109,10 +109,14 @@ export default function createManagedSubscriptionClient(
...
       }
+      if (!operation.text && !operation.id) {
+        throw Error("Neither subscription query nor id was provided.");
+      }
       const subscription = subscriptionClient
         .request({
           operationName: operation.name,
-          query: operation.text!,
+          // Use id as query when not available e.g. when using persisted queries.
+          query: (operation.text || operation.id)!,
```
Comment (alinhle) [inline src/core/client/framework/lib/network/createManagedSubscriptionClient.ts]: Add a `pq:` prefix.

File: tsconfig.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "target": "es5",
+    "target": "ES2018",
```
Comment (nick-funk) [inline tsconfig.json:3]: If I remember correctly, this is setting the transpile output target for TS transpiler.

Was this intentional? Jumping from ES5 (2009) to ES2018 seems like a big compatibility ask.

**Review by nick-funk — COMMENTED** [review]
Submitting with comment, I've reviewed and all looks good, just have a question about the tsconfig target change.

File: tsconfig.json (hunk 2 above)
Comment (wyattjoh) [reply tsconfig.json:3]: This is only on the server code, as we already depend on node >= 8, we have no issues here, the advantage is that we can now iterate over ES6 `Map`, before we couldn't ;)

Comment (nick-funk) [reply tsconfig.json:3]: Sweeeeeet!

**Review by nick-funk — APPROVED** [review]

**Pull request merged**